src/store/helium/heliumStatusSlice.ts

```diff
@@ -1,67 +1,85 @@
+import { getIncidents } from '../../utils/StatusClient'
+import { Incident, StatusInfo } from './StatusTypes'
 
-export type HeliumStatusState = {
-  status: {
-    page?: StatusInfo
-    status?: Status
-  }
-  incidents: {
-    page?: StatusInfo
-    incidents?: Incident[]
-  }
-}
-const initialState: HeliumStatusState = {
-  status: {},
-  incidents: {},
+const STATUS_KEY = 'statusKey'
+const ONBOARDING_COMPONENT_ID = '74lkk1qwp5xq'
+const APP_COMPONENT_ID = 'qfcq2xt6v0xm'
+const COMPONENT_IDS = [ONBOARDING_COMPONENT_ID, APP_COMPONENT_ID]
+const INCIDENT_STATUSES = [
+  'investigating',
+  'identified',
+  'scheduled',
+  'in_progress',
+  'verifying',
+]
+
+type StoredIncident = {
+  id: string
+  status: string
 }
 
-const getAlertDescription = (timestamp: string) => {
-  return `Last updated ${formatDistanceToNow(new Date(timestamp), {
-    locale: shortLocale,
-    addSuffix: true,
-  })}. Tap for info.`
+export type HeliumStatusState = {
+  page?: StatusInfo
+  incidents: Incident[]
 }
 
-export const fetchStatus = createAsyncThunk<{
-  page: StatusInfo
-  status: Status
-}>('heliumStatus/fetchStatus', async () => getStatus())
+const initialState: HeliumStatusState = {
+  incidents: [],
+}
 
-export const fetchIncidents = createAsyncThunk<{
-  page: StatusInfo
-  incidents: Incident[]
-}>('heliumStatus/fetchIncidents', async () => getIncidents())
+export const fetchIncidents = createAsyncThunk(
+  'heliumStatus/fetchIncidents',
+  async () => {
+    let storedIncidents: StoredIncident[] = []
+    const storedIncidentsStr = await AsyncStorage.getItem(STATUS_KEY)
+    if (storedIncidentsStr) {
+      storedIncidents = JSON.parse(storedIncidentsStr) as StoredIncident[]
+    }
+    const incidents: {
+      page: StatusInfo
+      incidents: Incident[]
+    } = await getIncidents()
+    return {
+      storedIncidents,
+      ...incidents,
+    }
+  },
+)
 
 const heliumStatusSlice = createSlice({
   name: 'heliumStatus',
   initialState,
   reducers: {},
   extraReducers: (builder) => {
-    builder.addCase(fetchStatus.fulfilled, (state, { payload }) => {
-      state.status = payload
-      if (payload?.status?.indicator && payload?.status?.indicator !== 'none') {
-        showMessage({
-          type: payload.status.indicator === 'critical' ? 'danger' : 'warning',
-          message: payload.status.description,
-          description: getAlertDescription(payload.page.updated_at),
-        })
-      }
-    })
     builder.addCase(fetchIncidents.fulfilled, (state, { payload }) => {
-      state.incidents = payload
-      const lastIncident = payload?.incidents[0]
-      if (lastIncident?.impact && lastIncident?.impact !== 'none') {
-        showMessage({
-          type: lastIncident.impact === 'critical' ? 'danger' : 'warning',
-          message: lastIncident.name,
-          description: getAlertDescription(lastIncident.updated_at),
-        })
-      }
+      const incidents = payload.incidents.flatMap((incident) => {
```

Comment:
I haven't seen `flatMap` used this way before. it certainly works. you could also use `reduce` and conditionally add to the accumulator

Reply:
This was actually a trick I found in the Elixir hex docs a little while back. I kinda like it 🤷

Reply:
fair enough, it works :)